src/utils/fiatToCrypto.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import querystring from 'querystring';
+import {
+  RAMPNETWORK_WIDGET_URL,
+  RAMPNETWORK_API_KEY,
+  SENDWYRE_WIDGET_URL,
+  SENDWYRE_ACCOUNT_ID,
+  SENDWYRE_RETURN_URL,
+} from 'react-native-dotenv';
+
+import type { AltalixTrxParams } from 'models/FiatToCryptoProviders';
+import type SDKWrapper from 'services/api';
+
+export function rampWidgetUrl(address: string, email?: string) {
+  const params = {
+    hostApiKey: RAMPNETWORK_API_KEY,
+    userAddress: address,
+    ...(email ? {} : { userEmailAddress: email }),
+  };
+
+  return `${RAMPNETWORK_WIDGET_URL}?${querystring.stringify(params)}`;
+}
+
+export function wyreWidgetUrl(address: string) {
+  return `${SENDWYRE_WIDGET_URL}?${querystring.stringify({
+    accountId: SENDWYRE_ACCOUNT_ID,
+    dest: `ethereum:${address}`,
+    redirectUrl: SENDWYRE_RETURN_URL,
+  })}`;
+}
+
+export function altalixWidgetUrl(params: AltalixTrxParams, api: SDKWrapper) {
+  return api.generateAltalixTransactionUrl(params);
```

Comment:
It's not a state change, but an address to be used only once -- it would add artificial complexity to a simple 1. construct url 2. open browser; action. Unless that's a solved problem somewhere else in the app?

Reply:
I haven't seen anything similar that's why it caught my eye.  The most similar case I can think of would be in lending actions, for example `calculateLendingDepositTransactionEstimateAction`. We use reducer for storing the fact that we're calculating transaction fee and the transaction fee even though we use it only once on one screen and it doesn't need to be in redux store really. And as far as I understand (it's a bit hackish) you're in fact creating an action that returns (?) the api and that's weird to me.

Reply:
I looked at that action -- the resulting reducer fragment is still passed to a component, whereas here it's given to a function (indirectly to `InAppBrowser.open`) when a button is pressed. You're correct, it's a hack, and I'd like to have a better solution, but it's needed because the api wrapper is accessible only from within actions while having a method that returns something that doesn't belong to the state. What I wanted was to avoid breaking up the code between a click handler and a `componentDidUpdate`. (By _only once_ I meant not that it occurs in a single screen/component, but that it's used immediately after construction and should be discarded after.)

Reply:
Ok, let's leave it 👍